third_party/move/move-model/src/ty.rs

```diff
@@ -28,14 +28,14 @@ pub enum Type {
     Primitive(PrimitiveType),
     Tuple(Vec<Type>),
     Vector(Box<Type>),
-    Struct(ModuleId, StructId, Vec<Type>),
+    Struct(ModuleId, StructId, /*type-params*/ Vec<Type>),
     TypeParameter(u16),
+    Fun(/*args*/ Box<Type>, /*result*/ Box<Type>),
 
     // Types only appearing in programs.
     Reference(ReferenceKind, Box<Type>),
 
-    // Types only appearing in specifications
-    Fun(Box<Type>, Box<Type>),
```

Comment:
We should also update https://github.com/vineethk/aptos-core/blob/577472b4cf119439cafab47f19298a32ed5b858c/third_party/move/move-model/src/ty.rs#L327?

Reply:
+1

Reply:
Nice find! Addressed, PLTA.